File: src/main/java/com/example/edumanager/entities/Turma.java
package com.example.edumanager.entities;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.Year;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Entity
public class Turma {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String nome;

    private Year ano;

    private String periodo;

    private String turno;

    private String sala;

    private String disciplina;

    private EnumStatusTurma status = EnumStatusTurma.ABERTA;

    // FK de turmaDisciplina
}

File: front/app/types/turma.ts
export class Turma{

    constructor(
        public id: number | null,
        public nome: string,
        public ano: number,
        public periodo: string,
        public turno: string,
        public sala: string,
        public disciplina: string,
        public status: string,)
        {        

    }
}